src/activation_token/wayland.rs

```diff
@@ -0,0 +1,59 @@
+use wayland_client::{protocol::wl_surface::WlSurface, Proxy, QueueHandle};
+use wayland_protocols::xdg::activation::v1::client::{
+    xdg_activation_token_v1::{Event, XdgActivationTokenV1},
+    xdg_activation_v1::XdgActivationV1,
+};
+
+#[derive(Debug, Default)]
+pub struct WaylandActivationToken {
+    pub token: String,
+    // We use a option to transform Self into Gtk4ActivationToken.
+    pub(super) inner: Option<XdgActivationTokenV1>,
+}
+
+impl Drop for WaylandActivationToken {
+    fn drop(&mut self) {
+        if let Some(wl_token) = self.inner.take() {
+            wl_token.destroy();
+        }
+    }
+}
+
+impl WaylandActivationToken {
+    pub fn from_surface(surface: &WlSurface) -> Result<Self, Box<dyn std::error::Error>> {
+        let cnx = wayland_client::Connection::connect_to_env().unwrap();
+        let wl_activation = XdgActivationV1::from_id(&cnx, surface.id()).unwrap();
```

Comment:
This look like it will always panic, ID of a WlSurface will never be equal to ID of XdgActivation object.